eclipse-collections-code-generator/src/main/resources/impl/set/mutable/primitiveHashSet.stg

```diff
@@ -1610,7 +1615,7 @@ public class <name>HashSet extends Abstract<name>Set implements Mutable<name>Set
             {
                 removeValue = <name>HashSet.this.table[this.position - 1];
             }
-            <name>HashSet.this.remove(removeValue);
+            <name>HashSet.this.remove(removeValue, false);
```

Comment:
While this is good enough to fix the bug, I think there's a better implementation that we should add to the backlog.

We should basically inline the call to remove here, removing the call to rehash, and directly mutating the backing table. When we delegate to remove(), we'll be calling probe() again. We already know the position in the table, because we're iterating through positions and have the position field.

This would also be more consistent with UnifiedSet.

This MR doesn't seem to address the same bug in primitive maps, where they delegate to removeKey and rehash.

Reply:
I don't see a rehash in the maps. I checked 7.x, 8.x and 9.x